File: testdata/bench/kotlin/BenchmarkTest00023.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.request.*
import io.ktor.server.response.*
import io.ktor.http.*

// Vulnerable: XSS via form data reflected in HTML
suspend fun handler00023(call: ApplicationCall) {
    val params = call.receiveParameters()
    val comment = params["comment"]
    call.respondText("<div class='comment'>$comment</div>", ContentType.Text.Html)
}

File: testdata/bench/javascript/BenchmarkTest00023.js
// Vulnerable: XSS via template literal in res.send
const express = require('express');
const router = express.Router();
router.get('/test023', (req, res) => {
    const comment = req.body.comment;
    res.send(`<div class="comment">${comment}</div>`);
});
module.exports = router;
